test/SimpleWebRequest.spec.ts

```diff
@@ -92,7 +106,158 @@ describe('SimpleWebRequest', () => {
         expect(
             () => new SimpleWebRequest<string>(url, method, {}, () => headers).start()
         ).toThrowError(`Don't set Content-Type with options.headers -- use it with the options.contentType property`)
+        test_resetQueues();
     })
 
+    describe('blocking', () => {
+        let maxRequests = 0;
+
+        beforeEach(() => {
+            maxRequests = SimpleWebRequestOptions.MaxSimultaneousRequests;
+            SimpleWebRequestOptions.MaxSimultaneousRequests = 0;
+            jasmine.clock().install();
+        });
+        afterEach(() => {
+            SimpleWebRequestOptions.MaxSimultaneousRequests = maxRequests; 
+            jasmine.clock().uninstall();
+        });
+
+        it('executes the requests by priority and age', () => {
+            const url = faker.internet.url();
+            const method = 'GET';
+            const onSuccess1 = jasmine.createSpy('onSuccess1');
+            const onSuccess2 = jasmine.createSpy('onSuccess2');
+            const onSuccess3 = jasmine.createSpy('onSuccess3');
+            const onSuccess4 = jasmine.createSpy('onSuccess4');
+            const status = 200;
+    
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Low }).start().then(onSuccess1);
```

Comment:
Suggestion: Use `onSuccessN` in the expected order, not the created order. It is harder to follow that you expect 2->4->1->3.